spec/amp-shadow-doc.md

```diff
@@ -96,6 +96,42 @@ Both `AMP.attachShadowDoc` and `AMP.attachShadowDocAsStream` return a `ShadowDoc
 - `shadowDoc.close()` - closes the AMP document and frees the resources.
 
 
+## Shadow DOM API and polyfills
+
+AMP Shadow Docs rely heavily on [Shadow DOM API](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Shadow_DOM). This is a powerful and elegant API, part of Web Components family. It allows natural isolation between major parts of the page and, as such, an ideal tool for PWAs and AMP Shadow Docs.
+
+Shadow DOM is currently implemented in Chrome and newer Safari, so it has a solid modern support and definitely not ubiquitous. AMP Shadow Docs API polyfills the necessary parts of Shadow DOM.
+
+However, not all advanced Shadow DOM features are polyfilled by AMP Shadow Docs. In particular, [shadow slots](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Slot) are not polyfilled. If you'd like to use slots and similar advanced features, please use one of the Shadow DOM polyfill, such as [WebComponents.js](https://github.com/webcomponents/webcomponentsjs). If you do, we recommend the following code structure (using Web Components polyfill as an example):
```

Comment:
I remember that we were worried that the developers may try to use their own or another less reputable polyfill for Shadow DOM slots, and it may conflict with our current implementation. 

I think it would be good to mention that the webcomponents script was the polyfill that we tested for `pwa.js` and we recommend as we can confirm that it works. I feel like the wording suggests that any polyfill will work (which it definitely should), but it would be better for someone trying AMP Shadow DOM for the first time.

Reply:
I made notes that we tested with Web Components polyfill and invited to report other polyfills. We'll consider including docs for others depending on popularity and quality.